walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,79 @@
 
 package com.google.sps;
 
+import java.util.*; 
 import java.util.Collection;
+import com.google.sps.TimeRange;
+import com.google.common.collect.Iterables;
 
+/** 
+ * Determines a list of available time slots to schedule a meeting given
+ * already scheduled meetings. 
+ */
 public final class FindMeetingQuery {
+  
+  /** 
+   * Returns the open windows for which {@code request} can be scheduled in
+   * while taking account conflicting meetings in {@code events}.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> requiredAttendees = request.getAttendees();
+    Collection<TimeRange> availableTimes = assembleTime(events, requiredAttendees, request.getDuration());
+    return adjustAvailableTimes;
+  }
+
+  /**
+   * Returns the open windows for which {@code request} can be scheduled in,
+   * after making {@code events} TimeRanges unavailable in {@code availableTimes}.
```

Comment:
in your comment you should never reference internal code. So you should omit mention of availableTimes.

You can rephrase as: ... after making {@code events} TimeRanges unavailable for scheduling.

Reply:
Fixed